### ISSUE TITLE
createSagas function

### ISSUE THREAD

**Issue by TaranbirBhullar** [reduxjs/redux-toolkit#511]
Just a huge shout out to the team behind redux-toolkit. It has greatly helped with reducing boilerplate for our actions/reducers/store.

In our team we heavily use saga(s) to handle a lot of our asynchronous calls.
A few hiccups we encounter as we write saga(s) today. 

1. We have to use `createAction` to create actions for saga. It would be nice to have actions created in a similar way that `createSlice`.
2. Using `PayloadAction` within a generator, is not very type specific, so we use `.match` make sure that we are typed correctly. It would be nice to have the strict typing available always.
3. This is more related to `redux-saga`, it's when we need to combine sagas, we would like it that there was a way to wrap them in their right effect and return a list that we can yield directly. If others have found a way to do this better. Would love to have a discussion there.

Here's a proposal for a `createSagas` API, that will work well for our scenario. 

```
example = createSagas({
   name: string,
   sagas: { [T: string]: (action: ActionType) => Generator }
})
```
It would return something like

```
{
   actions: Object<string, ActionCreator>,
   sagas: Object<string, Generator>,
}
```

Benefit we have here is similar to createAction

**Comment by markerikson** [comment]
Appreciate the suggestion, but [we do not plan to add any specific support for sagas in Redux Toolkit](https://blog.isquaredsoftware.com/2020/02/blogged-answers-why-redux-toolkit-uses-thunks-for-async-logic/).

You're always welcome to write custom additional abstractions for your own specific needs.

**Comment by phryneas** [comment]
Hey there, great to hear that this is helping you :)

Some thoughts:

1. You can also create empty actions for use in your sagas in `createSlice`:
```ts
createSlice({
  name: "foo",
  reducers: {
    actionForMySaga(_, action: PayloadAction<number>){} // method body can stay empty
  }
})
```
2. Could you please give an example on what you want to do and why it doesn't work?
3. I'll go with @markerikson on this one. RTK neither encourages nor discourages the use of saga - that abstraction would better be in another package.

**Comment by cyberhck** [comment]
@TaranbirBhullar I didn't use redux-saga like the function collection they documented, here's a sample of how I use it: github.com/crazyfactory/ts-react-boilerplate (basically we make them a react class, and then pass any sdk client through constructor) (boilerplate directly imports a file, but if you wanna know how to do via sdk: https://github.com/fossapps/Feature.Manager.Ui/blob/master/src/app/sagas/rootSaga.ts (on root saga, we initialize the client only once, and pass to everything,), all sagas extends base saga: https://github.com/fossapps/Feature.Manager.Ui/blob/master/src/app/sagas/BaseSaga.ts

That way all sagas get a instance of sdk client and can use it. (At least that works for me,) how did you end up doing it?
